src/expr/src/vector_op/arithmetic_op.rs

```diff
@@ -327,6 +326,23 @@ where
     r.mul_float(l).ok_or(ExprError::NumericOutOfRange)
 }
 
+#[function("sqrt(float64) -> float64")]
+pub fn sqrt_f64(expr: F64) -> Result<F64> {
+    if !expr.is_finite() {
+        return Err(ExprError::InvalidParam {
+            name: "sqrt input",
+            reason: "input cannot be nan or infinity".to_string(),
+        });
+    }
```

Comment:
Good to take `nan` and `inf` into consideration, but actually PostgreSQL allows these:
```
test=# select sqrt('nan');
 sqrt 
------
  NaN
(1 row)

test=# select sqrt('inf');
   sqrt   
----------
 Infinity
(1 row)

test=# select sqrt('-0');
 sqrt 
------
   -0
(1 row)
```